assets/javascripts/issue_badge.js

```diff
@@ -1,38 +1,62 @@
 // call issue_badge
-function load_badge(url) {
-    $.ajax({
-        url: url,
-        async: true,
-        type: 'get'
-    }).done(function( html ) {
-        if ($(html).filter('div#issue_badge').length) {
-            $('#loggedas').after(html);
-        }
-    });
-}
+/* eslint-disable no-unused-vars */
+/* eslint-env jquery */
+const loadBadge = (url) => {
+  let request = new window.XMLHttpRequest()
+  request.open('GET', url, true)
 
-function display_badge_contents(url) {
-    $.ajax({
-        url: url,
-        async: true,
-        type: 'get'
-    }).done(function( html ) {
-        $('#link_issue_badge').after(html);
-    });
+  request.onload = () => {
+    if (request.status >= 200 && request.status < 400) {
+      let html = request.response
+      if (html.length > 0) {
+        $('#loggedas').after(html)
+      }
+    } else {
+      console.log('[IssueBadge] LoadBadge Response Error. Status: ' + request.status)
+    }
+  }
+
+  request.onerror = () => {
+    console.log("[IssueBadge] LoadBadge Request Error. Can't get badge data.")
+  }
+  request.send()
 }
 
-function change_badge_location() {
-    if (window.matchMedia( '(max-width: 899px)' ).matches) {
-        $('#quick-search').prepend($('#issue_badge'));
+const displayBadgeContents = (url) => {
+  let request = new window.XMLHttpRequest()
+  request.open('GET', url, true)
+
+  request.onload = () => {
+    if (request.status >= 200 && request.status < 400) {
+      let html = request.response
+      if (html.length > 0) {
+        document.getElementById('link_issue_badge').insertAdjacentHTML('afterend', html)
+      }
     } else {
-        $('#loggedas').after($('#issue_badge'));
+      console.log('[IssueBadge] DisplayBadge Error. Status: ' + request.status)
     }
+  }
+
+  request.onerror = () => {
+    console.log("[IssueBadge] DisplayBadge Request Error. Can't get badge data.")
+  }
+  request.send()
 }
 
-$(document).click(function(event) {
-    // Hide if badge_contents exists.
-    if (!$(event.target).is("#issue_badge_contents")) {
-        $("#issue_badge_contents").hide();
-        $("#issue_badge_contents").remove();
-    }
-});
+const changeBadgeLocation = () => {
+  const issueBadgeElement = document.getElementById('issue_badge')
+  if (window.matchMedia('(max-width: 899px)').matches) {
```

Comment:
jQuery’s prepend / append can evaluate JavaScript inside the response string.
I have to change this.